server/api/cvrs.py

```diff
@@ -66,6 +67,53 @@ def set_contest_metadata_from_cvrs(contest: Contest):
             choice.num_votes += choice_metadata["num_votes"]
 
 
+def all_cvr_choice_names_match(contest):
+    choice_names = {choice.name for choice in contest.choices}
+    return all(
+        choice_names.issubset(
+            jurisdiction.cvr_contests_metadata[contest.name]["choices"].keys()
+        )
+        for jurisdiction in contest.jurisdictions
+    )
+
+
+class HybridNumVotes(typing.NamedTuple):
+    num_votes_cvr: int
+    num_votes_non_cvr: int
+
+
+# For Hybrid audits, we need to compute the vote counts for the CVRs
+# specifically so we can subtract them from the total vote count and get the
+# vote count for the non-CVR ballots.
+def hybrid_contest_choice_vote_counts(
+    contest: Contest,
+) -> Optional[Dict[str, HybridNumVotes]]:
+    if not all_cvrs_uploaded(contest):
+        return None
+
+    # TODO raise an error in /sample-sizes
+    if not all_cvr_choice_names_match(contest):
+        return None
+
+    cvr_choice_votes = {choice.id: 0 for choice in contest.choices}
+    for jurisdiction in contest.jurisdictions:
+        cvr_contests_metadata = typing.cast(
+            JSONDict, jurisdiction.cvr_contests_metadata
+        )
+        contest_metadata = cvr_contests_metadata[contest.name]
+        for choice_name, choice_metadata in contest_metadata["choices"].items():
+            choice = next(c for c in contest.choices if c.name == choice_name)
```

Comment:
is `contest.choices` going to do a DB query every time? If so should we pull those once outside of the two `for` loops?

Reply:
Nope, once you load a relationship (e.g. `contest.choices`), it's just a Python object.